src/main/navigation/navigation.c

```diff
@@ -1698,24 +1715,29 @@ void setHomePosition(const fpVector3_t * pos, int32_t yaw, navSetWaypointFlags_t
     if ((useMask & NAV_POS_UPDATE_XY) != 0) {
         posControl.homePosition.pos.x = pos->x;
         posControl.homePosition.pos.y = pos->y;
+        posControl.homeFlags |= NAV_HOME_VALID_XY;
     }
 
     // Z-position
     if ((useMask & NAV_POS_UPDATE_Z) != 0) {
         posControl.homePosition.pos.z = pos->z;
+        posControl.homeFlags |= NAV_HOME_VALID_Z;
     }
 
     // Heading
     if ((useMask & NAV_POS_UPDATE_HEADING) != 0) {
         // Heading
         posControl.homePosition.yaw = yaw;
+        if (posControl.flags.estHeadingStatus >= EST_USABLE) {
```

Comment:
`setHomePosition` may be called with arbitrary `yaw`. We shouldn't check for `estHeadingStatus` in this function. We could check for valid heading before calling `setHomePosition` though.

Reply:
Opps. Nice catch! Thing is, for the best result we actually want to store it with the "fake" yaw, then correct it, since we don't don't have a valid yaw at arming time in crafts without mag. Maybe we can pass the `homeFlags` to `setHomePosition()`?

Reply:
Not sure, need to think about it. `setHomePosition` has access to homeFlags anyway

Reply:
I mean we can let the caller tell `setHomePosition()` which validity flags it should set for the home position.

Reply:
Ah, that's definitely a good idea